homeassistant/components/nibe_heatpump/config_flow.py

```diff
@@ -27,13 +27,22 @@
     LOGGER,
 )
 
+PORT_SELECTOR = vol.All(
+    selector.NumberSelector(
+        selector.NumberSelectorConfig(
+            min=1, step=1, max=65535, mode=selector.NumberSelectorMode.BOX
+        )
+    ),
+    vol.Coerce(int),
+)
+
 STEP_USER_DATA_SCHEMA = vol.Schema(
     {
         vol.Required(CONF_MODEL): vol.In(list(Model.__members__)),
-        vol.Required(CONF_IP_ADDRESS): str,
-        vol.Required(CONF_LISTENING_PORT, default=9999): cv.port,
-        vol.Required(CONF_REMOTE_READ_PORT, default=9999): cv.port,
-        vol.Required(CONF_REMOTE_WRITE_PORT, default=10000): cv.port,
+        vol.Required(CONF_IP_ADDRESS): selector.TextSelector(),
+        vol.Required(CONF_LISTENING_PORT, default=9999): PORT_SELECTOR,
+        vol.Required(CONF_REMOTE_READ_PORT, default=9999): PORT_SELECTOR,
+        vol.Required(CONF_REMOTE_WRITE_PORT, default=10000): PORT_SELECTOR,
```

Comment:
These changes would not bee needed if https://github.com/home-assistant/frontend/issues/14164 is resolved.